### PR TITLE
Raven db 13874 identities2

### PR DESCRIPTION
**Pull request by Danielle9897**
### Issue link
https://issues.hibernatingrhinos.com/issue/RavenDB-13874

### Additional description
Create Identities View

### Type of change
- New feature

### How risky is the change?
- Low 

### Backward compatibility
- Non-breaking change

### Is it platform specific issue?
- No

### Documentation update
- This change requires a documentation update. 

### Testing 
- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?
- No

### UI work
- No UI work is needed

### CODE REVIEW

File: src/Raven.Studio/typescript/commands/database/identities/setIdentityCommand.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import commandBase = require("commands/commandBase");
+import endpoints = require("endpoints");
+import database = require("models/resources/database");
+
+class setIdentityCommand extends commandBase {
```
Comment (ml054) [inline src/Raven.Studio/typescript/commands/database/identities/setIdentityCommand.ts]: set or seed?

File: src/Raven.Studio/wwwroot/Content/css/pages/identities.less [hunk 2]
```diff
@@ -0,0 +1,8 @@
+.identities-view {
+    
+    .identities-list-container {
+        width: 700px;
```
Comment (ml054) [inline src/Raven.Studio/wwwroot/Content/css/pages/identities.less:4]: can't we use classes from bootstrap ?

File: src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+        
+        this.initObservables();
+        this.initValidation();
+    }
+    
+    private initObservables(): void {
+        this.prefixWithPipe = ko.pureComputed(() => {
+            let prefix = this.prefix();
+            
+            if (prefix.length >= 1 && prefix.charAt(prefix.length - 1) !== "|") {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts]: there is endsWith method on string.

File: src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+                prefix += "|";
+            }
+            
+            return prefix;
+        });
+        
+        this.prefixWithoutPipe = ko.pureComputed(() => {
+            let prefix = this.prefix();
+            
+            if (prefix.length >=2 && prefix.charAt(prefix.length - 1) === "|") {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts]: if (prefix.endsWith("|"))

File: src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+        });
+        
+        this.validationGroup = ko.validatedObservable({
+            name: this.prefix,
+            value: this.value
+        })
+    }
+    
+    static empty() {
+        return new identity("", null);
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts:89]: maybe we want default value here? (0)?

File: src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+    }
+
+    addNewIdentity($event: JQueryEventObject): void {
+        eventsCollector.default.reportEvent("identity", "new");
+        this.isNewIdentity(true);
+        this.editedIdentityItem(identity.empty())
+
+        this.editedIdentityItem().prefix.subscribe(() => {
+            const item = this.editedIdentityItem();
+            item.prefixAlreadyExists(!!this.identityPrefixList.find(x => x.toLocaleLowerCase() === item.prefixWithPipe().toLocaleLowerCase() ||
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts]: this.identityPrefixList Can this list contain items with `|` and w/o `|`?

Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts]: fixed - using only the `prefixWithoutPipe` condition ...

File: src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts (hunk 4 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts]: done

File: src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts (hunk 3 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts]: done

File: src/Raven.Studio/typescript/commands/database/identities/setIdentityCommand.ts (hunk 1 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/commands/database/identities/setIdentityCommand.ts]: done

File: src/Raven.Studio/wwwroot/Content/css/pages/identities.less (hunk 2 above)
Comment (Danielle9897) [reply src/Raven.Studio/wwwroot/Content/css/pages/identities.less:4]: leaving this - as discussed

File: src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts (hunk 5 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts:89]: leaving this - as discussed

File: src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts [hunk 7]
```diff
@@ -46,7 +46,7 @@ class identity {
         this.prefixWithoutPipe = ko.pureComputed(() => {
             let prefix = this.prefix();
             
-            if (prefix.length >=2 && prefix.charAt(prefix.length - 1) === "|") {
+            if (prefix.length >=2 && prefix.endsWith("|")) {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts]: I think we don't need first check, right?

File: src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts [hunk 8]
```diff
@@ -36,7 +36,7 @@ class identity {
         this.prefixWithPipe = ko.pureComputed(() => {
             let prefix = this.prefix();
             
-            if (prefix.length >= 1 && prefix.charAt(prefix.length - 1) !== "|") {
+            if (prefix.length >= 1 && !prefix.endsWith("|")) {
```
Comment (ml054) [inline src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts]: do we need first check?

Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts]: don't want to add pipe for null prefix so **don't** want to use just:
`if (!prefix.endsWith("|")) {`

refactored to:
```
if (prefix.length && !prefix.endsWith("|")) {
    prefix += "|";
}
```

Comment (ml054) [reply src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts]: then `if (prefix && !prefix.endsWith("|"))`

File: src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts (hunk 7 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts]: ok - removed first check - 
and also fixed the `prefixAlreadyExists` subscribe

File: src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts (hunk 8 above)
Comment (Danielle9897) [reply src/Raven.Studio/typescript/viewmodels/database/identities/identities.ts]: done

**Review by ml054 — APPROVED** [review]

**Pull request merged**